### PR TITLE
spawners: introduce AVOCADO_SPAWNER_EGG

### LINKED ISSUE #5072: Allow the deployment and use of development versions of Avocado
**Is your feature request related to a problem? Please describe.**
Epic issue https://github.com/avocado-framework/avocado/issues/4934 covers aspects such as creating and publishing (AFAICT) packages for every Avocado release, for every Python version, and how they will be deployed and installed on `DeploymentSpawner`s.

But, this doesn't address the development time needs, where unreleased code may be needed in both the job side (say running out of a GIT checkout), and on the isolated spawner environment.

**Describe the solution you'd like**
There should be a way to signal a preference for given wheels to be used, instead of the latest release ones. 

**Describe alternatives you've considered**
Building the wheels is probably going to be a separate step (such as the current `make whee`, and It will probably be necessary to refer to them by a local (non HTTP(s)) uri.   Bonus point for an all-in-one `--mirror-this-repo-as-a-wheel-and-deploy-it` kind of feature, but I think this will be too complex.

### PR DESCRIPTION
**Pull request by beraldoleal**
When using an isolated spawner (i.e: Podman), sometimes we need to test our
local branch on the isolated environment side. By default, Avocado will try to
download the matching eggs from our Git repository (or use the cached version).
But this will not apply your local changes at this point. You can build
your local egg with `python setup.py bdist_egg` and use the result as
input for this variable so Avocado will skip the downloading process and
use the local egg.

Fixes #5072, #4934.

Signed-off-by: Beraldo Leal <bleal@redhat.com>

### CODE REVIEW

**Review by richtja — COMMENTED** [review]
LGTM, I found just small typo.

**Comment by clebergnu** [comment]
I'm trying to understand why this isn't simply a configuration item?  It'd use a self documenting mechanism, and other users should be able to refer to alternative eggs themselves (maybe hosted remotely). IMO, this is similar to `--podman-spawner-image` where users get to select which image to use, so they can select which avocado egg to use.

I understand this is born out of out development needs, but I can't see why this won't be a "first class" feature.

**Comment by beraldoleal** [comment]
> I'm trying to understand why this isn't simply a configuration item? It'd use a self documenting mechanism, and other users should be able to refer to alternative eggs themselves (maybe hosted remotely). IMO, this is similar to `--podman-spawner-image` where users get to select which image to use, so they can select which avocado egg to use.
> 
> I understand this is born out of out development needs, but I can't see why this won't be a "first class" feature.

Hi @clebergnu, Sorry for the delay here.

I tough about implementing as you suggested at the time. However after a while I decided for this approach because I could not find a good use case where *users* would need to change the egg (differently of the spawner-image). Since this is covering a "development time need", maybe we could reduce the number of command line options, making our command line simpler by "hiding" those settings from end-users.

If you think this could be useful for end-users, then I'm fine with the change.

Makes any sense to you?

(PS: I'm strongly against your proposal, just trying to make things a little bit easier for users here)

**Comment by clebergnu** [comment]
> > I'm trying to understand why this isn't simply a configuration item? It'd use a self documenting mechanism, and other users should be able to refer to alternative eggs themselves (maybe hosted remotely). IMO, this is similar to `--podman-spawner-image` where users get to select which image to use, so they can select which avocado egg to use.
> > I understand this is born out of out development needs, but I can't see why this won't be a "first class" feature.
> 
> Hi @clebergnu, Sorry for the delay here.
> 
> I tough about implementing as you suggested at the time. However after a while I decided for this approach because I could not find a good use case where _users_ would need to change the egg (differently of the spawner-image). Since this is covering a "development time need", maybe we could reduce the number of command line options, making our command line simpler by "hiding" those settings from end-users.
> 

I think there may be use cases for "regular users" to choose a different egg... Even if someone is not developing Avocado, there may be the need to deploy a slightly different version. But, the essence of my suggestion is that:

1. we already have a standard and self documenting configuration system
2. as a framework, we should give introduce more mechanisms and less policies

And if growing the command line itself is deemed to be a problem, we also already deal with that by having configuration items that are controlled by the configuration file, but do not have an equivalent command line switch.

> If you think this could be useful for end-users, then I'm fine with the change.
> 

I think it would, and just by being using the standard configuration system, it'd improve user's experience.

> Makes any sense to you?
> 
> (PS: I'm strongly against your proposal, just trying to make things a little bit easier for users here)

I assume you meant *not* strongly against my proposal. ;)

**Comment by beraldoleal** [comment]
> I assume you meant not strongly against my proposal. ;)

Of course! :)

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @beraldoleal,

The changes look good (thanks!) but the documentation is now inaccurate.  Given that, by using `settings` there's automatic generated documentation, I don't think it'd be a problem to remove the hand written doc section (but it's up to you).

**Comment by beraldoleal** [comment]
@clebergnu fixed. thanks.

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @beraldoleal,

The change in `docs/source/guides/contributor/index.rst` is not needed anymore. Other than that, LGTM.

File: avocado/plugins/spawners/podman.py [hunk 1]
```diff
@@ -49,6 +49,16 @@ def initialize(self):
             help_msg=help_msg,
             default=default_distro)
 
+        help_msg = ('Avocado EGG path to be used during initial boostrap '
```
Comment (ana) [inline avocado/plugins/spawners/podman.py:52]: bootstrap*

**Comment by ana** [comment]
The commit title says "introduce hability" that should be "introduce (the) ability".
Also please, consider changing  "Avocado EGG" for "Avocado egg" in the help message. It's parsed as a some acronym instead of a name.

**Comment by ana** [comment]
Hi @beraldoleal,
this looks good to me, just missing this from @clebergnu :

> The change in `docs/source/guides/contributor/index.rst` is not needed anymore. Other than that, LGTM.

**Comment by beraldoleal** [comment]
> Hi @beraldoleal, this looks good to me, just missing this from @clebergnu :
> 
> > The change in `docs/source/guides/contributor/index.rst` is not needed anymore. Other than that, LGTM.

I removed the content and forgot to remove this from the index. Just did with the force-push. Thanks.

**Review by ana — APPROVED** [review]

**Review by clebergnu — APPROVED** [review]
Thanks for the updates @beraldoleal, LGTM!

**Pull request merged**